indexer/bag_indexer/src/test/scala/uk/ac/wellcome/platform/archive/indexer/bag/BagIndexerFeatureTest.scala

```diff
@@ -39,4 +40,62 @@ class BagIndexerFeatureTest
     )
 
   override val mapping: MappingDefinition = BagsIndexConfig.mapping
+
+  val version: BagVersion = BagVersion(1)
+  val ingest: Ingest = createIngestWith(version = Some(version))
+  val pipelineContext: PipelineContext = PipelineContext(ingest)
+
+  val bagInfo = createBagInfoWith(externalIdentifier = ingest.externalIdentifier)
+
+  val storageManifest: StorageManifest = createStorageManifestWith(
+    ingestId = ingest.id,
+    space = ingest.space,
+    version = version,
+    bagInfo = bagInfo
+  )
+  val payload: KnownReplicasPayload = createKnownReplicasPayloadWith(
+    context = pipelineContext,
+    version = version
+  )
+
+  override def createT: (KnownReplicasPayload, String) = {
+    val bagId = BagId(
+      space = payload.storageSpace,
+      externalIdentifier = payload.externalIdentifier
+    )
+
+    (payload, bagId.toString)
+  }
+
+  override def convertIndexedT(
+                                payload: KnownReplicasPayload
+                              ): IndexedStorageManifest = {
+    IndexedStorageManifest(storageManifest)
+  }
+
+  override def withIndexerWorker[R](index: Index, queue: SQS.Queue)(
+    testWith: TestWith[
+      IndexerWorker[KnownReplicasPayload, StorageManifest, IndexedStorageManifest],
+      R
+    ]
+  )(implicit decoder: Decoder[KnownReplicasPayload]): R = {
```

Comment:
nit: Do we actually need to pass this decoder? Don't we get it by importing `JsonUtil._`?

Reply:
I think as this is overriding an existing function, we must define this.